rpc/client.go

```diff
@@ -113,34 +111,39 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 	}
 	clientMu.Unlock()
 
-	go c.connect(opts, context)
+	go func() {
+		if err := c.connect(opts, context); err != nil {
+			log.Errorf("client %s failed to connect: %v", c.addr, err)
+			c.Close()
+		}
+	}()
 	return c
 }
 
 // connect dials the connection in a backoff/retry loop.
-func (c *Client) connect(opts *retry.Options, context *Context) {
+func (c *Client) connect(opts *retry.Config, context *Context) error {
 	// Attempt to dial connection.
 	retryOpts := clientRetryOptions
 	if opts != nil {
 		retryOpts = *opts
 	}
-	retryOpts.Tag = fmt.Sprintf("client %s connection", c.addr)
 	retryOpts.Stopper = context.Stopper
 
-	err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
+	for r := retry.Start(retryOpts); r.Next(); {
```

Comment:
I like the for-loop approach. It makes it clearer to me that this is looping. Wondering if `Next` should be renamed to `WaitNext` to make it clearer that it is pausing.